MetovaTestKit/CellTesting/MTKCollectionViewTestCell.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  MTKTableViewTestCell.swift
+//  MetovaTestKit
+//
+//  Created by Nick Griffith on 2018-12-14
+//  Copyright © 2018 Metova. All rights reserved.
+//
+//  MIT License
+//
+//  Permission is hereby granted, free of charge, to any person obtaining
+//  a copy of this software and associated documentation files (the
+//  "Software"), to deal in the Software without restriction, including
+//  without limitation the rights to use, copy, modify, merge, publish,
+//  distribute, sublicense, and/or sell copies of the Software, and to
+//  permit persons to whom the Software is furnished to do so, subject to
+//  the following conditions:
+//
+//  The above copyright notice and this permission notice shall be
+//  included in all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+//  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+//  MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+//  NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+//  LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+//  OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+//  WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+import XCTest
+
+extension UICollectionView {
+    
+    /// This method attempts to produce a collection view cell from the collection view at the specified index path.  It then attempts to cast the cell to the specified type.  If either of these two fail, this function will produce a test failure and the test block will not be executed.  If both pass, the resulting cell, cast to the specified type, will be passed into the testBlock where the developer can run further assertions on the cell.
+    ///
+    /// - Parameters:
+    ///   - cellType: Type of cell to cast to before passing into test block.
+    ///   - indexPath: IndexPath to grab a cell from.
+    ///   - file: The file the test is called from.
+    ///   - line: The line the test is called from.
+    ///   - testBlock: A block of code containing tests to run.  This block is not guaranteed to execute.  If it does not execute, this method will fail the test.
+    /// - Throws: Rethrows any error thrown by the test block.  Does not throw any errors on its own.
+    func testCell<T: UICollectionViewCell>(
+        cellType: T.Type = T.self,
```

Comment:
I would recommend moving this argument after the `at indexPath: IndexPath` argument, as arguments with default values are typically moved toward the end of the argument list. I would also recommend adding the `as` function label to the front for more fluent usage.

```swift
testCell(at: indexPath, as: MyCustomTableViewCell.self) { cell in 
    XCTAssertEqual(cell.customValue, 1)
}
```